lib/buildingsync/model_articulation/building.rb

```diff
@@ -280,61 +301,129 @@ def get_system_type
-          epw_file_name = "CZ03RV2.epw"
-        elsif climate_zone == "4A" || climate_zone == "4B" || climate_zone == "4C"
-          epw_file_name = "CZ04RV2.epw"
-        elsif climate_zone == "5A" || climate_zone == "5B" || climate_zone == "5C"
-          epw_file_name = "CZ05RV2.epw"
-        elsif climate_zone == "6A" || climate_zone == "6B"
-          epw_file_name = "CZ06RV2.epw"
-        elsif climate_zone == "7"
-          epw_file_name = "CZ07RV2.epw"
-        elsif climate_zone == "8"
-          epw_file_name = "CZ08RV2.epw"
+        set_weather_and_climate_zone_from_climate_zone(climate_zone, standard_to_be_used, latitude, longitude)
+      end
+
+      # setting the current year, so we do not get these annoying log messages:
+      # [openstudio.model.YearDescription] <1> 'UseWeatherFile' is not yet a supported option for YearDescription
+      yearDescription = @model.getYearDescription
+      yearDescription.setCalendarYear(::Date.today.year)
+
+      # add final condition
+      OpenStudio.logFree(OpenStudio::Info, 'BuildingSync.Facility.set_weather_and_climate_zone', "The final weather file is #{@model.getWeatherFile.city} and the model has #{@model.getDesignDays.size} design day objects.")
+    end
+
+    def set_weather_and_climate_zone_from_climate_zone(climate_zone, standard_to_be_used, latitude, longitude)
+
+      climate_zone_standard_string = climate_zone
+      OpenStudio.logFree(OpenStudio::Info, 'BuildingSync.Facility.set_weather_and_climate_zone_from_climate_zone', "climate zone: #{climate_zone}")
+      if standard_to_be_used == CA_TITLE24 && !climate_zone.nil?
+        climate_zone_standard_string = "CEC T24-CEC#{climate_zone.gsub('Climate Zone', '').strip}"
+      elsif standard_to_be_used == ASHRAE90_1 && !climate_zone.nil?
+        climate_zone_standard_string = "ASHRAE 169-2006-#{climate_zone.gsub('Climate Zone', '').strip}"
+      elsif climate_zone.nil?
+        climate_zone_standard_string = ''
+      end
+
+      if !$open_studio_standards.nil? && !$open_studio_standards.model_add_design_days_and_weather_file(@model, climate_zone_standard_string, nil)
+        OpenStudio.logFree(OpenStudio::Error, 'BuildingSync.Building.set_weater_and_climate_zone', "Cannot add design days and weather file for climate zone: #{climate_zone}, no epw file provided")
+      end
+
+      # overwrite latitude and longitude if available
+      if !latitude.nil? || !longitude.nil?
+        site = @model.getSite
+        if !latitude.nil?
+          site.setLatitude(latitude.to_f)
+        end
+        if !longitude.nil?
+          site.setLongitude(longitude.to_f)
+        end
+      end
+
+      weather_file = @model.getWeatherFile
+
+      OpenStudio.logFree(OpenStudio::Info, 'BuildingSync.Facility.set_weater_and_climate_zone', "city is #{weather_file.city}. State is #{weather_file.stateProvinceRegion}")
+
+      # Set climate zone
+      climateZones = @model.getClimateZones
```

Comment:
Make the climate_zone its own method and write tests around it.

Reply:
done